workspaces/diff-engine/src/queries/endpoint.rs

```diff
@@ -16,9 +19,20 @@ impl<'a> EndpointQueries<'a> {
       endpoint_projection,
     }
   }
+
+  pub fn extract_normalized_path(interaction: &HttpInteraction) -> &str {
+    if interaction.request.path.eq("/") {
+      &interaction.request.path
+    } else if interaction.request.path.ends_with("/") {
+      &interaction.request.path[..interaction.request.path.len() - 1]
+    } else {
+      &interaction.request.path
+    }
+  }
   pub fn resolve_path(&self, interaction: &HttpInteraction) -> Option<PathComponentIdRef> {
+    let path = Self::extract_normalized_path(interaction);
     // eprintln!("{}", path);
-    let mut path_components = interaction.request.path.split('/');
+    let mut path_components = path.split('/');
```

Comment:
Rather than solving this by string matching and manipulation, wouldn't we be better off skipping any last components that are empty, post-split? We do a skip for the first empty one, so it seems within expectation we might do the same for the last?

Reply:
unless I'm missing something with Rust, it's more complex to track whether you are on the last item in the split iterator than to normalize it beforehand. Skipping the first item in an iterator is simpler afaik. In addition, the normalization has meaning as an extracted function. It is used in a few places in the scala and typescript code IIRC.

Reply:
Fair enough, was probably thinking about that with JS brain where it would have been very straightforward. Not to say it would be very difficult in Rust, but must admit I can't instantly produce how to do that either!